ddtrace/debugging/_origin/span.py

```diff
@@ -196,12 +196,44 @@ def __exit__(self, exc_type, exc_value, traceback):
 
 
 @dataclass
-class SpanCodeOriginProcessor(SpanProcessor):
+class SpanCodeOriginProcessorEntry:
     __uploader__ = LogsIntakeUploaderV1
 
-    _instance: t.Optional["SpanCodeOriginProcessor"] = None
+    _entry_instance: t.Optional["SpanCodeOriginProcessorEntry"] = None
     _handler: t.Optional[t.Callable] = None
 
+    @classmethod
+    def enable(cls):
+        if cls._entry_instance is not None:
+            return
+
+        cls._entry_instance = cls()
+
+        # Register code origin for span with the snapshot uploader
+        cls.__uploader__.register(UploaderProduct.CODE_ORIGIN_SPAN)
+
+        # Register the entrypoint wrapping for entry spans
+        cls._handler = handler = partial(wrap_entrypoint, cls.__uploader__.get_collector())
+        core.on("service_entrypoint.patch", handler)
+
+    @classmethod
+    def disable(cls):
+        if cls._entry_instance is None:
+            return
+
+        # Unregister the entrypoint wrapping for entry spans
+        core.reset_listeners("service_entrypoint.patch", cls._handler)
+        # Unregister code origin for span with the snapshot uploader
+        cls.__uploader__.unregister(UploaderProduct.CODE_ORIGIN_SPAN)
+
+        cls._handler = None
+        cls._entry_instance = None
+
+
+@dataclass
+class SpanCodeOriginProcessor(SpanCodeOriginProcessorEntry, SpanProcessor):
```

Comment:
Since we are effectively breaking up code origin into entry and exit, I would maybe
```suggestion
class SpanCodeOriginProcessorExit(SpanProcessor):
```
and write logic to enable them independently in the, e.g. the product module. WDYT?

Reply:
> For the uploader would you create a new product?

If there is no functional need for that, I would just use one uploader product for both.